test/runtests.jl

```diff
@@ -1,13 +1,46 @@
 using Nabla
-using Test, LinearAlgebra, Statistics, Random
-using Distributions, BenchmarkTools, SpecialFunctions, DualNumbers
+using Test, LinearAlgebra, Statistics, Random, ForwardDiff
+using Distributions, BenchmarkTools, SpecialFunctions
 
 using Nabla: unbox, pos, tape, oneslike, zeroslike
 
 # Helper function for comparing `Ref`s, since they don't compare equal under `==`
 ref_equal(a::Ref{T}, b::Ref{T}) where {T} = a[] == b[]
 ref_equal(a::Ref, b::Ref) = false
 
+# for comparing against scalar rules
+derivative_via_frule(f, x) = last(Nabla.frule((Nabla.NO_FIELDS, 1.0), f, x))
+# Sensiblity checkes that his is defined right
+@test derivative_via_frule(cos, 0) == 0
+@test derivative_via_frule(sin, 0) == 1
+@test derivative_via_frule(sin, 1.2) == derivative_via_frule(sin, 2π + 1.2)
+
+# These are the core scalar sensitives Nabla expects to have defined
+# we test against them both for sensitives/scalar.jl and in sensitivities/functional.jl
+
+const UNARY_SCALAR_SENSITIVITIES = [
+    # Base:
+    +, -, abs, abs2, acos, acosd, acosh, acot, acotd, acoth, acsc, acscd, acsch, asec,
+    asecd, asech, asin, asind, asinh, atand, atanh, cbrt, cos, cosd, cosh, cospi, cot,cotd,
+    coth, csc, cscd, csch, deg2rad, exp, exp10, exp2, expm1, inv, log, log10, log2,
+    rad2deg, sec, secd, sech, sin, sind, sinh, sinpi, sqrt, tan, tand, tanh, transpose,
+    # SpecialFunctions.jl:
+    airyai, airyaiprime, airybi, airybiprime, besselj0, besselj1, bessely0, bessely1,
+    dawson, digamma, erf, erfc, erfcinv, erfcx, erfi, erfinv, gamma, invdigamma, loggamma,
+    trigamma,
+]
+
+const BINARY_SCALAR_SENSITIVITIES = [
+    # Base:
+    *, +, -, /, \, ^, hypot, max, min,
+    # SpecialFunctions.jl:
+    besseli, besselj, besselk, bessely, beta, polygamma,
```

Comment:
Could just merge arrays here with the below since all `ONLY_DIFF_IN_SECOND_ARG_SENSITIVITIES` are present here, but nbd if you don't want to.

Reply:
It us an intentional subset of the above
because we have a bunch of loops that are like

```
for f in BINARY_SCALAR_SENSITIVIES
    if f in ONLY_DIFF_IN_SECOND_ARG_SENSITIVITIES
         # test only second
    else
         # test both
    end
end
```

I could define
```julia
const DIFF_IN_BOTH_ARG_SENTIVITIES = ....
const BINARY_SCALAR_SENSITIVITIES = union(DIFF_IN_BOTH_ARG_SENTIVITIES, ONLY_DIFF_IN_SECOND_ARG_SENSITIVITIES)
```
Do you think that would be cleaner?

Reply:
Honestly either way works, just suggested this as it could potentially cut down on some lines of code.